monolithic/monolithic-layered/src/test/java/com/epam/training/microservices/monolithic/web/DiseaseControllerIT.java

```diff
@@ -0,0 +1,65 @@
+package com.epam.training.microservices.monolithic.web;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.redirectedUrl;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+
+/**
+ * Test {@link DiseaseController}
+ */
+public class DiseaseControllerIT extends MockMvcTest {
+
+    @Test
+    @DisplayName("Test viewAll endpoint for Diseases")
+    public void shouldReturnAllDiseasesList() throws Exception {
+        MockHttpServletRequestBuilder viewAllReq = get("/diseases");
+        this.mockMvc.perform(viewAllReq)
+                .andExpect(status().isOk())
+                // TODO uncomment after fix of #8
+                // .andExpect(content().string(containsString("<title>The Drug System - Diseases</title>")))
+                .andExpect(content().string(containsString(">Diseases</h1>")))
+                .andExpect(content().string(containsString("<a href=\"/diseases/1\"")));
+    }
+
+    @Test
+    @DisplayName("Test viewSingle endpoint for Diseases")
+    public void shouldReturnSpecificDiseaseDetails() throws Exception {
+        MockHttpServletRequestBuilder viewSingleReq = get("/diseases/1");
+        this.mockMvc.perform(viewSingleReq)
+                .andExpect(status().isOk())
+                // TODO uncomment after fix of #8
+                // .andExpect(content().string(containsString("<title>The Drug System - Edit Disease</title>")))
+                .andExpect(content().string(containsString(">Edit Disease</h1>")))
+                .andExpect(content().string(containsString("Asperger syndrome")))
```

Comment:
HTMLUnit is a great option for assertions like this because it allows to check that the element is not only present in the HTML but also it is placed in the right place.

Reply:
I don't know how to integrate HTMLUnit with Spring MockMVC. I don't want to start a real web server, because I think it will make tests slower.

1. The official integration between Spring MVC Test and HTMLUnit is outdated. See https://mvnrepository.com/artifact/org.springframework.test.htmlunit/spring-test-htmlunit which was last updated May 14, 2015.
2. When I tried to simply use `final HtmlPage page = webClient.getPage("http://localhost/diseases")` as written in the [official documentation](https://www.htmlunit.org/gettingStarted.html), I got error "org.htmlunit.FailingHttpStatusCodeException: 404 Not Found for http://localhost/diseases".

There's nobody to respond to requests sent by HTMLUnit web client. Because I'm using [MockMVC](https://docs.spring.io/spring-framework/reference/testing/spring-mvc-test-framework.html) in tests:

> It performs full Spring MVC request handling but via mock request and response objects instead of a running server.

Reply:
Sure, leave it to me, when have time I'll add integration with it.